src/components/engineService.ts

```diff
@@ -440,7 +443,17 @@ export class EngineService {
 
 		const noraRustEngineExecutableUri = this.#noraRustEngineExecutableUri;
 
+		let timer: NodeJS.Timeout | null = null;
+
 		interfase.on('line', async (line) => {
+			if (timer !== null) {
+				clearTimeout(timer);
+			}
+
+			timer = setTimeout(() => {
+				childProcess.kill();
```

Comment:
what happens when we attempt to kill an already terminated process?

Reply:
just tried to call `kill` twice
```
childProcess.kill();
childProcess.kill();
```
second one was just ignored